superset-frontend/src/explore/components/controls/SelectControl.jsx

```diff
@@ -124,6 +124,17 @@ export default class SelectControl extends React.PureComponent {
       onChangeVal = val[valueKey];
     }
     this.props.onChange(onChangeVal, []);
+
+    if (!this.state.options.some(option => option.value === onChangeVal)) {
+      this.setState(previousState => {
+        const newOptions = [
+          ...previousState.options,
+          { value: onChangeVal, label: onChangeVal },
+        ];
+        newOptions.sort((a, b) => a.value - b.value);
+        return { options: newOptions };
+      });
+    }
```

Comment:
@aobapirak can you confirm the need for this? To my understanding as long as the `allowNewOptions` for the Select is true, the component will allow for the creation of new options without having to rely on custom state?

Reply:
The new option doesn't stick to the options, i.e. whenever I enter a new option it will disappear when I add a new option, except for refreshing the page it will show all new options from this code:
`mapStateToProps: ({ controls }: ControlPanelState) => ({ choices: controls?.server_page_length_options?.value, })`

So the code above I created to allow SelectControl can be sticked new option to the options and sort them.

Reply:
@aobapirak you are right that the options are not persisted unless selected. I had a different memory of this piece of the Select component. There is a problem of introducing this behaviour in the SelectControl wrapper as there is a risk of introducing inconsistency. Would you be open to have a wider discussion to make this a default behaviour of the Select component instead?